From 392e808ebae97faecdaadf044c7ebd112b82cc94 Mon Sep 17 00:00:00 2001
From: cinerea0 <cinerea0@disroot.org>
Date: Mon, 11 Dec 2023 14:51:53 -0500
Subject: [PATCH] fix: EntryData size test passes on 32-bit (#197)

EntryData doesn't need to be exactly 80 to test memory consumption, just
less than it.
---
 src/traverse.rs | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/traverse.rs b/src/traverse.rs
index 157ce2c..b2c8724 100644
--- a/src/traverse.rs
+++ b/src/traverse.rs
@@ -312,10 +312,10 @@ mod tests {
 
     #[test]
     fn size_of_entry_data() {
-        assert_eq!(
-            std::mem::size_of::<EntryData>(),
-            80,
-            "the size of this should not change unexpectedly as it affects overall memory consumption"
+        assert!(
+            std::mem::size_of::<EntryData>() <= 80,
+            "the size of this ({}) should not exceed 80 as it affects overall memory consumption",
+            std::mem::size_of::<EntryData>()
         );
     }
 }
-- 
2.43.0

